src/ClassFileLocator.php

```diff
@@ -112,15 +112,17 @@ public function accept()
                             }
                             continue;
                         }
-                        list($type, $content, $line) = $token;
+                        list($type, $content) = $token;
                         switch ($type) {
                             case T_STRING:
                             case T_NS_SEPARATOR:
+                            case 312: // T_NAME_FULLY_QUALIFIED
+                            case 314: // T_NAME_QUALIFIED
```

Comment:
this is not gonna work, these constant can have different values on different PHP version, see for example:
https://3v4l.org/XFk83

I think, what we need here, is to define these constants if not defined - but with string values, something what has been done in PHP_CodeSniffer. Then it's  gonna be just placeholder on older PHP versions and should behave as expects on PHP 8.0+

I think to do that, we would need separate autoload.php file, add it to composer and define the constants there.

Reply:
Hi @michalbundyra,

You're right but I'm not sure to understand fully what you propose. 
Do you have an example to help me implement your proposition ?

Reply:
@codisart Not sure if I get this right, but if we need those tokens on PHP 8.0 only, you could add something like:

```php

if (PHP_VERSION_ID > 80000) {
    switch ($type) {
        T_NAME_FULLY_QUALIFIED:
        T_NAME_QUALIFIED:
            $namespace .= $content;
         break;
    }
}
```

This would use the constants when PHP version is greather than PHP 8.0.0 🤷‍♂️

Reply:
I see that php cs used the same method https://github.com/squizlabs/PHP_CodeSniffer/pull/3063/files

Reply:
I really hope I am not creating a vicious technical debt. ^^

Reply:
As there is no other way to properly support this, I think there is no other way.
Even tho, there should be already a unit test for covering this. 🤷🏼‍♂️